dali/operators/reader/loader/video/frames_decoder_test.cc

```diff
@@ -103,6 +107,140 @@ TEST_F(FramesDecoderTest, VariableFrameRate) {
     this->CompareFrames(frame.data(), this->GetVfrFrame(1, 0), file.FrameSize());
 }
 
+TEST_F(FramesDecoderTest, VariableFrameRateGpu) {
+    ASSERT_TRUE(cuInitChecked());
+    CUdevice device = { 0 };
+    CUDA_CALL(cuDeviceGet(&device, 0));
+    CUcontext context = { 0 };
+    CUDA_CALL(cuCtxCreate(&context, 0, device));
+
+
+    std::string path = testing::dali_extra_path() + "/db/video/vfr/test_2.avi";
+
+    // Create file, build index
+    FramesDecoderGpu file(path);
+
+    ASSERT_EQ(file.Height(), 600);
+    ASSERT_EQ(file.Width(), 800);
+    ASSERT_EQ(file.Channels(), 3);
+    ASSERT_EQ(file.NumFrames(), 60);
+
+    uint8_t *frame = nullptr;
+    cuMemAlloc(
+        reinterpret_cast<CUdeviceptr*>(&frame),
+        file.FrameSize());
+    std::vector<uint8_t> frame_cpu(file.FrameSize());
```

Comment:
Why not use `DeviceBuffer` and just `resize` and `from_device`?

Reply:
Done